src/page/ThemePage.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { Banner } from '@/components/common/Banner/Banner';
+import Footer from '@/components/Footer';
+import { Header } from '@/components/Header';
+import GoodsList from '@/components/ThemeComponents/GoodsList';
+
+import type { ThemeKey } from './theme.value';
+import { THEME_HEADERS } from './theme.value';
+
+const ThemePage = () => {
+  const { themeKey } = useParams<{ themeKey: string }>();
+  const [bannerProps, setBannerProps] = useState(THEME_HEADERS.Birthday);
+
+  useEffect(() => {
+    if (themeKey && THEME_HEADERS[themeKey as ThemeKey]) {
+      setBannerProps(THEME_HEADERS[themeKey as ThemeKey]);
+    } else {
+      setBannerProps(THEME_HEADERS.Default);
+    }
+  }, [themeKey]);
```

Comment:
```suggestion
const ThemePage = () => {
  const { themeKey } = useParams<{ themeKey: ThemeKey }>();
  const [bannerProps, setBannerProps] = useState(THEME_HEADERS.Birthday);

  useEffect(() => {
    if (themeKey && THEME_HEADERS[themeKey]) {
      setBannerProps(THEME_HEADERS[themeKey]);
    } else {
      setBannerProps(THEME_HEADERS.Default);
    }
  }, [themeKey]);
```

themeKey가 언제나 ThemeKey 라는 것이 보장된다면 위와 같이 작성할 수 있을 것 같아요-!